src/main/java/org/dsa/iot/calendar/Actions.java

```diff
@@ -0,0 +1,306 @@
+                        clientSecret = event.getParameter("clientSecret").getString();
+                    }
+
+                    NodeBuilder calendarBuilder = superRoot.createChild(desc);
+                    calendarBuilder.setAttribute("type", new Value("google"));
+                    calendarBuilder.setRoConfig("clientId", new Value(clientId));
+                    calendarBuilder.setRoConfig("clientSecret", new Value(clientSecret));
+                    Node calendarNode = calendarBuilder.build();
+
+                    NodeBuilder eventsBuilder = calendarNode.createChild("events");
+                    eventsBuilder.setDisplayName("Events");
+                    Node events = eventsBuilder.build();
+
+                    try {
+                        GoogleCalendar cal = new GoogleCalendar(calendarNode, clientId, clientSecret);
+                        calendars.put(desc, cal);
+
+                        Actions.addCreateEventNode(calendarNode);
+                        Actions.addRemoveCalendarNode(calendarNode);
+                        Actions.addRefreshCalendarNode(calendarNode);
+                        cal.attemptAuthorize(calendarNode);
+                    } catch (GeneralSecurityException | IOException e) {
+                        e.printStackTrace();
+                    }
+                }
+            });
+            addParameter(new Parameter("desc", ValueType.STRING));
+            addParameter(new Parameter("clientId", ValueType.STRING));
+            addParameter(new Parameter("clientSecret", ValueType.STRING));
+        }
+    }
+
+    private static class RemoveAccount extends Action {
+        public RemoveAccount() {
+            super(Permission.CONFIG, new Handler<ActionResult>() {
+                @Override
+                public void handle(ActionResult event) {
+                    Node calendar = event.getNode().getParent();
+                    event.getNode().getParent().getParent().removeChild(calendar);
+                }
+            });
+        }
+    }
+
+    private static class RefreshBuilder extends Action {
+        public RefreshBuilder(final BaseCalendar calendar) {
+            super(Permission.WRITE, new Handler<ActionResult>() {
+                @Override
+                public void handle(ActionResult event) {
+                    calendar.updateCalendar();
+                }
+            });
+        }
+    }
+
+    private static class CreateEvent extends Action {
+        public CreateEvent(final BaseCalendar calendar) {
+            super(Permission.WRITE, new Handler<ActionResult>() {
+                @Override
+                public void handle(ActionResult actionResult) {
+                    String title = actionResult.getParameter("title").getString();
+                    if (title != null && !title.isEmpty()) {
+                        String desc = actionResult.getParameter("desc", new Value("")).getString();
```

Comment:
I think it'd be a good idea to extract all those Parameters/Node names into constants. It'll be much easier to refactor and rename things.